### PR TITLE
Build Jenkins master image on a schedule

### PR DESCRIPTION
**Pull request by fwilhe**
This PR replaces the automated build (and test) of Jenkins master on Docker hub. It also schedules a daily build and push (if smoke test succeeds) of the latest tag so we provide an up-to-date Jenkins master with updated plugins. 

The goals of this is to provide a more secure jenkins environment.

A secondary goal is to migrate off Docker hub as a build environment as it caused issues with non-triggered builds in the past.

### CODE REVIEW

**Comment by fwilhe** [comment]
I figured one potential weakness: We're running the full test suite on travis, so it might occur that the smoke test succeeds, we push to docker hub but the test on travis fails due to some plugin incompatibility or similar issues.

Options
* Somehow check if travis succeeded before pushing (not sure how we can do that)
* Also run the full-blown infra tests on github before we push (will be a larger PR)
* Accept this risk and mitigate it when something bad happened (With the "backup of latest" tag this should still be possible)

I think I'm ok with option 3. Anyone disagrees?

**Comment by fwilhe** [comment]
Hi @marcusholl, @CCFenner, @nevskrem et al, reviews would be appreciated. I've motivated this change above in the PR description, but of course I don't want to merge this prematurely so please take a critical look if you find some time.

Thanks.

File: .github/workflows/push-release.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+    - uses: actions/checkout@master
+    - name: Build and smoke-test a release candidate of the Docker image
+      run: |
+        docker build jenkins-master --tag ppiper/jenkins-master:RC
+        docker run --name cx-jenkins-master-test --detach --publish 8080:8080 \
+          -v /var/run/docker.sock:/var/run/docker.sock \
+          ppiper/jenkins-master:RC
+        # Wait up to 5 minutes for Jenkins to be up and fail if it does not return 200 after this time
+        docker exec cx-jenkins-master-test timeout 300 bash -c "until curl --fail http://localhost:8080/api/json; do sleep 5; done"
```
Comment (daniel-kurzynski) [inline .github/workflows/push-release.yml]: So we won't push if it is not up after 5min. But how to we make sure that the image is created later. Is re-triggering working?

Comment (fwilhe) [reply .github/workflows/push-release.yml]: You could delete and re-push the tag. You can also restart failed workflows in actions manually.

In the worst case, we don't push a broken image and that given tag has no built image. I think this is acceptable.

**Comment by fwilhe** [comment]
> From reading the code it looks good to me. Not fully happy with the cron schedule (only once a day at 6 am, that means under the day i won't see an update but that is a minor thing).

No, this does also trigger on push

```
on:
  push:
    branches:
      - "**"
  schedule:
    - cron:  '0 6 * * *'
```

The docker push will only happen on the master branch. The idea of the cron is to provide updated images even if no merge happens (which is most of the time in this repo anyway)

**Pull request merged**